src/main/java/org/truffleruby/language/yield/YieldExpressionNode.java

```diff
@@ -97,4 +107,18 @@ private YieldNode getYieldNode() {
         return yieldNode;
     }
 
+    private void warnIfYieldInSingletonClass(boolean isSingletonClass) {
+        if (isSingletonClass) {
+            if (warnNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                final WarnNode newWarnNode = new WarnNode();
```

Comment:
Is this extra local variable needed? Otherwise I'd remove it to be more concise.